wagtail/contrib/wagtailroutablepage/tests.py

```diff
@@ -185,3 +185,15 @@ def test_templatetag_reverse_external_view(self):
                               'external_view', 'joe-bloggs')
 
         self.assertEqual(url, self.routable_page.url + 'external/joe-bloggs/')
+
+    def test_templatetag_reverse_external_view_without_append_slash(self):
+        import wagtail.wagtailcore.models
+        wagtail.wagtailcore.models.WAGTAIL_APPEND_SLASH = False
```

Comment:
We should use ``mock.patch()`` for this. Otherwise, if the ``routablepageurl`` call raises an exception, ``WAGTAIL_APPEND_SLASH`` will not be restored to ``True`` which could lead to confusing test breakages later on.

Reply:
I updated my PR to use `mock`